### PR TITLE
Siyao h/1.17/windows

### PR DESCRIPTION
**Pull request by SiyaoHuang**

### CODE REVIEW

File: engine/Build.bee.cs [hunk 1]
```diff
@@ -624,6 +628,110 @@ static void SetupDependency(NativeProgram np)
...
+            "-DFLUTTER_JIT_RUNTIME=1",
+
+            "-DSK_ENABLE_SPIRV_VALIDATION",
+            "-D_CRT_SECURE_NO_WARNINGS",
+            "-D_HAS_EXCEPTIONS=0",
+            "-DWIN32_LEAN_AND_MEAN",
+            "-DNOMINMAX",
+            "-DSK_GAMMA_APPLY_TO_A8",
+            "-DSK_ALLOW_STATIC_GLOBAL_INITIALIZERS=1",
+            // "-DGR_TEST_UTILS=1",
```
Comment (zhuxingwei) [inline engine/Build.bee.cs:689]: we should add some comments on why not using this Macro, maybe paste this link (https://github.com/microsoft/vcpkg/issues/12123) as a reference
and add a TODO to find a way to fix this issue by changing some configurations on txt_lib

File: engine/Build.bee.cs [hunk 2]
```diff
@@ -645,6 +753,30 @@ static void SetupDependency(NativeProgram np)
             "-fvisibility-inlines-hidden",
         }));
 
+        var basePath = skiaRoot + "/out/Debug";
```
Comment (zhuxingwei) [inline engine/Build.bee.cs:756]: the name "basePath" is confusion. change this to something better?

File: engine/README.md [hunk 3]
```diff
@@ -6,56 +6,13 @@ This is the engine code of UIWidgets.
...
-```
-Ignore this error: "lld-link: error: could not open 'EGL': no such file or directory"
-
-convert icudtl.dat to object file in skia
-```
-cd SkiaRoot/third_party/externals/icu/flutter/
-ld -r -b binary -o icudtl.o icudtl.dat
-```
-### Build flutter fml
+### Build flutter txt
```
Comment (zhuxingwei) [inline engine/README.md:15]: can we rename the sub-title to "Build Dependencies" ?

File: engine/README.md [hunk 4]
```diff
@@ -75,40 +32,94 @@ gclient sync -D
...
 ```
 set GYP_MSVS_OVERRIDE_PATH=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community
 cd engine/src
 python ./flutter/tools/gn --unoptimized
-ninja -C .\out\host_debug_unopt\ flutter/fml:fml_lib
+ninja -C out\host_debug_unopt flutter/third_party/txt:txt_lib
 ```
 
-### Create symbolic
+### Build Skia
```
Comment (zhuxingwei) [inline engine/README.md:87]: so we build skia just for icudtl.dat ? if so, maybe we should rename it to "Build icu data" or something better?

Comment (SiyaoHuang) [reply engine/README.md:87]: This build is for getting egl.dll, egl.dll.lib, eglv2.dll, eglv2.dll.lib as well. Icu data can bee generated directly

**Pull request merged**